From 0375c5d840ef3b9be8952ef86d62ad4947e765a7 Mon Sep 17 00:00:00 2001
From: Christoph Haag <haagch+git@frickel.club>
Date: Mon, 9 Jan 2017 01:06:53 +0100
Subject: [PATCH] remove pointless boost check

See
https://github.com/OSVR/OSVR-Core/pull/495#issue-184905680

"Is this version check still worth doing? Or can we just blacklist the known-bad version(s)?"

It was only ever a problem for one or two old boost versions that were bad
---
 cmake-local/BoostTargets.cmake    | 6 ------
 src/osvr/Common/IPCRingBuffer.cpp | 8 --------
 2 files changed, 14 deletions(-)

diff --git a/cmake-local/BoostTargets.cmake b/cmake-local/BoostTargets.cmake
index f4b72059..9913f616 100644
--- a/cmake-local/BoostTargets.cmake
+++ b/cmake-local/BoostTargets.cmake
@@ -23,12 +23,6 @@ else()
 endif()
 
 find_package(Boost ${OSVR_MIN_BOOST} COMPONENTS ${required_boost_components} REQUIRED)
-if(Boost_VERSION GREATER 106200)
-    # Current max code-reviewed version: Boost 1.62
-    # When this is updated - source code must also be updated!
-    message(SEND_ERROR "Using an unreviewed Boost version - inspect the Boost Interprocess release notes/changelog/diffs to see if any ABI breaks took place as they may affect client/server interoperability.")
-    message(SEND_ERROR "The corresponding source file to update is src/osvr/Common/IPCRingBuffer.cpp")
-endif()
 
 if(NEED_BOOST_THREAD)
     add_library(boost_thread INTERFACE)
diff --git a/src/osvr/Common/IPCRingBuffer.cpp b/src/osvr/Common/IPCRingBuffer.cpp
index 95150f77..fe871d6e 100644
--- a/src/osvr/Common/IPCRingBuffer.cpp
+++ b/src/osvr/Common/IPCRingBuffer.cpp
@@ -55,14 +55,6 @@ namespace common {
     /// that would interfere with communication.
     static IPCRingBuffer::abi_level_type SHM_SOURCE_ABI_LEVEL = 0;
 
-/// Some tests that can be automated for ensuring validity of the ABI level
-/// number.
-/// The base boost version test has been moved exclusively to CMake, to error
-/// out earlier.
-#if (BOOST_VERSION > 106200)
-#error                                                                         \
-    "Using an untested Boost version - inspect the Boost Interprocess release notes/changelog to see if any ABI breaks affect us."
-#endif
 
 #ifdef _WIN32
 #if (BOOST_VERSION < 105400)

base-commit: 7c69b804723b196e10b9c9818cd8801b2e06f6d3
-- 
2.11.0

